src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,172 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int DEFAULT_INDEX = 0;
+    private static final int DIVIDER = 2;
+    private Node<T> first;
+    private Node<T> last;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        Node<T> current = new Node<>(value);
+        if (first == null) {
+            first = current;
+        } else {
+            current.prevElement = last;
+            last.nextElement = current;
+        }
+        last = current;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        indexCheckerForAdding(index);
+        if (size() == DEFAULT_INDEX) {
+            add(value);
+            return;
+        }
+        if (index == DEFAULT_INDEX) {
+            Node<T> current = first;
+            first = new Node<>(value);
+            first.nextElement = current;
+            current.prevElement = first;
+            size++;
+            return;
+        }
+        if (index == size()) {
+            add(value);
+            return;
+        }
+        Node<T> current = getNodeByIndex(index);
+        Node<T> newNode = new Node<>(value);
+        newNode.prevElement = current.prevElement;
+        newNode.nextElement = current;
+        current.prevElement.nextElement = newNode;
+        current.prevElement = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexChecker(index);
+        return getNodeByIndex(index).value;
     }
```

Comment:
I think we should always call `indexChecker` method before `getNodeByIndex`,  
so can we move `indexChecker` inside `getNodeByIndex`?